include/fmt/format.h

```diff
@@ -51,7 +51,9 @@
 # define FMT_ICC_VERSION 0
 #endif
 
-#if (defined(__GNUC__) && !defined(__clang__) && __GNUC__ >= 406) || \
+#include "core.h"
+
+#if (defined(__GNUC__) && !defined(__clang__) && FMT_GCC_VERSION >= 406) || \
```

Comment:
You only need to check `FMT_GCC_VERSION` since it has a non-zero value iff `defined(__GNUC__) && !defined(__clang__)` is `true`